src/events.js

```diff
@@ -662,12 +662,15 @@ const handleCheckInModal = async (data, visitConcept, query) => {
         continueButtonText: "Continue to Specimen Link",
     };
 
-    const checkInOnCancel = () => {  };
+    const checkInOnCancel = () => {
+        dismissBiospecimenModal();
+      };
     const checkInOnContinue = async () => {
         try {
             const updatedResponse = await findParticipant(query);
             const updatedData = updatedResponse.data[0];
-    
+
+            dismissBiospecimenModal();
             specimenTemplate(updatedData);
         } catch (error) {
             showNotifications({ title: 'Error', body: 'There was an error checking in the participant. Please try again.' });
```

Comment:
Instead of splitting handlers of "Cancel" and "Continue" buttons in different locations, I suggest put these click handlers inside `showNotificationsCancelOrContinue`, to make things clearer.

Reply:
This function is used in multiple places. This will need a refactor in the future.